pkg/apis/management.cattle.io/v3/cluster_types.go

```diff
@@ -125,6 +125,13 @@ type ClusterSpecBase struct {
 	WindowsPreferedCluster                               bool                                    `json:"windowsPreferedCluster" norman:"noupdate"`
 	LocalClusterAuthEndpoint                             LocalClusterAuthEndpoint                `json:"localClusterAuthEndpoint,omitempty"`
 	ClusterSecrets                                       ClusterSecrets                          `json:"clusterSecrets" norman:"nocreate,noupdate"`
+	AgentDeploymentCustomization                         *AgentDeploymentCustomization           `json:"agentDeploymentCustomization,omitempty"`
```

Comment:
We want to define separate fields for cattle-cluster-agent and fleet-agent, as below:
```
ClusterAgentDeploymentCustomization *AgentDeploymentCustomization `json:"clusterAgentDeploymentCustomization,omitempty"`
FleetAgentDeploymentCustomization   *AgentDeploymentCustomization `json:"fleetAgentDeploymentCustomization,omitempty"`
```